src/roles/actors.rs

```diff
@@ -0,0 +1,356 @@
+        /// The member must then send a stake() transaction to approve the request and enter the desired role.
+        /// The account making the request will be bonded and must have
+        /// sufficient balance to cover the minimum stake for the role.
+        /// Bonding only occurs after successful entry into a role.
+        /// The request expires after REQUEST_LIFETIME blocks
+        RoleEntryRequests get(role_entry_requests) : Requests<T>;
+    }
+}
+
+decl_event! {
+    pub enum Event<T> where
+      <T as system::Trait>::AccountId {
+        Staked(AccountId, Role),
+    }
+}
+
+impl<T: Trait> Module<T> {
+    fn role_is_available(role: Role) -> bool {
+        Self::available_roles().into_iter().any(|r| role == r)
+    }
+
+    fn ensure_actor(role_key: &T::AccountId) -> Result<Actor<T>, &'static str> {
+        Self::actors_by_account_id(role_key).ok_or("not role key")
+    }
+
+    fn ensure_actor_is_member(role_key: &T::AccountId, member_id: MemberId<T>)
+        -> Result<Actor<T>, &'static str>
+    {
+        let actor = Self::ensure_actor(role_key)?;
+        if actor.member_id == member_id {
+            Ok(actor)
+        } else {
+            Err("actor not owned by member")
+        }
+    }
+
+    fn ensure_role_parameters(role: Role) -> Result<RoleParameters<T>, &'static str> {
+        Self::parameters(role).ok_or("no parameters for role")
+    }
+
+    // Mutating
+
+    fn remove_actor_from_service(actor_account: T::AccountId, role: Role, member_id: MemberId<T>) {
+        let accounts: Vec<T::AccountId> = Self::accounts_by_role(role)
+            .into_iter()
+            .filter(|account| !(*account == actor_account))
+            .collect();
+        <AccountsByRole<T>>::insert(role, accounts);
+
+        let accounts: Vec<T::AccountId> = Self::accounts_by_member(&member_id)
+            .into_iter()
+            .filter(|account| !(*account == actor_account))
+            .collect();
+        <AccountsByMember<T>>::insert(&member_id, accounts);
+
+        <ActorsByAccountId<T>>::remove(&actor_account);
+    }
+
+    fn apply_unstake(actor_account: T::AccountId, role: Role, member_id: MemberId<T>, unbonding_period: T::BlockNumber) {
+        // simple unstaking ...only applying unbonding period
+        <Bondage<T>>::insert(&actor_account, <system::Module<T>>::block_number() + unbonding_period);
+
+        Self::remove_actor_from_service(actor_account, role, member_id);
+    }
+}
+
+impl<T: Trait> Roles<T> for Module<T> {
+    fn is_role_account(account_id: &T::AccountId) -> bool {
+        <ActorsByAccountId<T>>::exists(account_id) || <Bondage<T>>::exists(account_id)
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        fn deposit_event<T>() = default;
+
+        fn on_initialise(now: T::BlockNumber) {
+            // clear expired requests
+            if now % T::BlockNumber::sa(DEFAULT_REQUEST_CLEARING_INTERVAL) == T::BlockNumber::zero() {
```

Comment:
Consider creating an extra variable just to make code easier to read, then you can get rid of a comment:
```suggestion
            let is_request_expired = now % T::BlockNumber::sa(DEFAULT_REQUEST_CLEARING_INTERVAL) == T::BlockNumber::zero()
            if is_request_expired {
```

Reply:
I see you point in general, but in this exact case the that check is just a check to see if we reached the interval where we actually check each request.